core/qbft/qbft.go

```diff
@@ -468,7 +469,11 @@ func isJustifiedRoundChange[I any, V Value[V]](d Definition[I, V], msg Msg[I, V]
 		return false
 	}
 
+	uniq := uniqSource[I, V]()
 	for _, prepare := range prepares {
+		if !uniq(prepare) {
```

Comment:
when counting messages, ensure that only one message per source/process is counted.